pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -142,61 +141,60 @@ func (m *Multicluster) close() (err error) {
 	for _, clusterID := range clusterIDs {
 		clusterID := clusterID
 		g.Go(func() error {
-			return m.ClusterDeleted(clusterID)
+			m.ClusterDeleted(clusterID)
+			return nil
 		})
 	}
-	err = g.Wait()
-	return
+	return g.Wait()
 }
 
 // ClusterAdded is passed to the secret controller as a callback to be called
 // when a remote cluster is added.  This function needs to set up all the handlers
 // to watch for resources being added, deleted or changed on remote clusters.
-func (m *Multicluster) ClusterAdded(cluster *multicluster.Cluster, clusterStopCh <-chan struct{}) error {
+func (m *Multicluster) ClusterAdded(cluster *multicluster.Cluster, clusterStopCh <-chan struct{}) {
 	m.m.Lock()
-	kubeController, kubeRegistry, options, configCluster, err := m.addCluster(cluster)
-	if err != nil {
+	kubeController, kubeRegistry, options, configCluster := m.addCluster(cluster)
+	if kubeController == nil {
```

Comment:
not sure i understand this. Seems an error for me

Reply:
This only happens when `m.closing`. Its not an error really, just no action needed. Definitely do not need to retry since its permanent

TBH I am suspicious that this case ever really happens anyways.

I added a comment in the code